app/controllers/patients_controller.rb

```diff
@@ -1,9 +1,15 @@
 # Create and update patients, plus the main patient view in edit
 class PatientsController < ApplicationController
+  before_action :confirm_user_has_data_access, only: [:index]
   before_action :find_patient, only: [:edit, :update]
   rescue_from Mongoid::Errors::DocumentNotFound,
               with: -> { redirect_to root_path }
 
+  def index
+    @patients = Patient.all
+    respond_to :csv
```

Comment:
Hm, I think `send_data` may only be for sending a binary stream (https://apidock.com/rails/ActionController/DataStreaming/send_data).  Is your main concern making sure we get the `Content-Disposition` header?